### PR TITLE
Watchlist endpoint

### PR DESCRIPTION
**Pull request by VinceWilliamsDev**
This endpoint allows users to create and manage a 'watchlist' of bridges they are interested in. 
Users can:
  POST to `/watchlist/:userID` to create a list.
  GET from `/watchlist/:userID` to view their list.
  PUT to `/watchlist/:userID` to update the notes and title fields.
  PUT to `/watchlist/bridge/:userID` to add bridges to their list.
  DELETE to `/watchlist/bridge/:userID` to remove bridges from their list.

Clearly documents the reason for the pull request and what has changed. Delete this test and add your own.
- [ ] This is a bugfix.
- [X] This is a new feature.
- [ ] This is part of a feature.

## Organization ##

- [ ] Have you linked this pull request to a Trello card?

## Review ##

- [X] Have you added two reviewers to this pull request?
- [ ] Is this a work in process?
- [X] Is this ready to be reviewed?

### CODE REVIEW

**Review by charisearter — APPROVED** [review]
Great workaround to the issues we were having on the database.

File: api/watchlist/watchlist-bridgesModel.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  bridges.forEach((bridge) => {
+    console.log(bridge);
+    bridge_array.push(bridge.project);
+  });
+
+  return bridge_array;
+}
+
+const addWatchlist = async (title, profile_id, notes, bridge_array) => {
+  // console.log(title, profile_id, notes, bridge_array);
```
Comment (Joe-Bay) [inline api/watchlist/watchlist-bridgesModel.js]: I would probably remove console logs like this

File: api/watchlist/watchlistRouter.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+// HTTP POST to create a new watchlist for a user
+
+router.post('/:id', authRequired, validateUserId, async function (req, res) {
+  const notes = String(req.body.notes);
+  const user = String(req.params.id);
+  const locations = req.body.locations;
+  const title = String(req.body.title);
+  console.log(notes, user, locations, title);
+
+  try {
```
Comment (Joe-Bay) [inline api/watchlist/watchlistRouter.js:33]: is this being used?

**Review by Joe-Bay — APPROVED** [review]
Just a few things I would remove but it looks really great otherwise.

File: api/watchlist/watchlist-bridgesModel.js (hunk 1 above)
Comment (VinceWilliamsDev) [reply api/watchlist/watchlist-bridgesModel.js]: Thanks, I've cleaned them up in the latest version

File: api/watchlist/watchlistRouter.js (hunk 2 above)
Comment (VinceWilliamsDev) [reply api/watchlist/watchlistRouter.js:33]: It was only for testing - it's been removed now.

**Pull request merged**